programs/lz4io.c

```diff
@@ -152,81 +155,108 @@ static const char* g_dictionaryFilename = NULL;
 /* ****************** Parameters ******************** */
 /* ************************************************** */
 
-int LZ4IO_setDictionaryFilename(const char* dictionaryFilename) {
-    g_dictionaryFilename = dictionaryFilename;
-    g_useDictionary = dictionaryFilename != NULL;
-    return g_useDictionary;
+LZ4IO_prefs_t* LZ4IO_defaultPreferences(void)
+{
+  LZ4IO_prefs_t* ret = malloc(sizeof(LZ4IO_prefs_t));
```

Comment:
`malloc()` can fail

Reply:
Good catch! Sorry about not catching that earlier.

Only question I have is which error number to throw, but we can sync up on that tomorrow morning :)